src/Renderer/Hal.php

```diff
@@ -222,13 +220,13 @@ protected function relToAbs($hal, $absoluteHrefs)
 	 */
 	public function addUriParameters($href, $absoluteHrefs)
 	{
+		$uri = Uri::getInstance($href);
+
 		if ($absoluteHrefs && substr($href, 0, 1) == '/')
 		{
-			$href = rtrim(JUri::base(), '/') . $href;
+			$href = rtrim($uri->base(), '/') . $href;
 		}
 
-		$uri = JUri::getInstance($href);
-
```

Comment:
Although $href now contains the correct absolute URL at this point, it isn't used to update $uri and so this method fails to convert relative URLs to absolute URLs.